dspy/streaming/streaming_listener.py

```diff
@@ -63,14 +67,19 @@ def receive(self, chunk: ModelResponseStream):
             end_identifier = self.json_adapter_end_identifier
 
             start_indicator = '"'
+        elif isinstance(settings.adapter, XMLAdapter):
+            start_identifier = self.xml_adapter_start_identifier
+            end_identifier = self.xml_adapter_end_identifier
+
+            start_indicator = "<"
         elif isinstance(settings.adapter, ChatAdapter) or settings.adapter is None:
             start_identifier = self.chat_adapter_start_identifier
             end_identifier = self.chat_adapter_end_identifier
 
             start_indicator = "["
         else:
             raise ValueError(
-                f"Unsupported adapter for streaming: {settings.adapter}, please use either ChatAdapter or "
+                f"Unsupported adapter for streaming: {settings.adapter}, please use either ChatAdapter, XMLAdapter or "
```

Comment:
nit: Can we define const for a list of adapters that support streaming?

Reply:
sg!